### ISSUE TITLE
SC2312 false positive inside conditionals 

### ISSUE THREAD

**Issue by bruce-ricard** [koalaman/shellcheck#2809]
#### For bugs
- Rule Id SC2312
- My shellcheck version: 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

[[ $(ls) ]]

```

#### Here's what shellcheck currently says:

```
In test.sh line 5:
[[ $(ls) ]]
     ^-- SC2312 (info): Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore).
```

#### Here's what I wanted or expected to see:

```
Your code is amazing!
```

**Comment by koalaman** [comment]
ShellCheck is right in this case. You have enabled `check-extra-masked-returns` and this construct does indeed mask the return value of the command. `[[ $(ls /tmp /nosuchfile) ]]` evaluates to true even though the command exits with failure.

**Comment by bruce-ricard** [comment]
Sorry, you are absolutely right. I got confused by the code I was reading.

For potential future readers, do NOT use `[[` mechanically after every `if`, `[[` is only to be able to do boolean manipulation like equals, or greater than etc. 
If you want to test if a function call like `ls` is successful, just run

```
if ls; then ... else ... fi
```
